OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -128,4 +128,17 @@ void Frame::extendRealizeTopology(SimTK::State& s) const
 {
     Super::extendRealizeTopology(s);
     groundTransformIndex = getCacheVariableIndex("ground_transform");
+}
+void Frame::extendRealizeVelocity(const SimTK::State& s) const
+{   
+    //Evaluate the lazy cache for the groundTransform so it doesn't get
+    //needlesly calculated by each one of the threads while realizing
+    //dynamics. (Also for thread-safety reasons)
+    Super::extendRealizeVelocity(s);
+
+    SimTK::Value<SimTK::Transform>::downcast(getSystem().getDefaultSubsystem().
+        updCacheEntry(s, groundTransformIndex)).upd() = calcGroundTransform(s);
+    
+    getSystem().getDefaultSubsystem().
+        markCacheValueRealized(s, groundTransformIndex);
```

Comment:
`Force::Gravity` is in Simbody? 

This might indicate a bug of some sort and we should dig deeper here. Another reason why this should be done at the position stage is that other position stage computations may depend on it.